db/migrate/20181024163850_add_source_type.rb

```diff
@@ -0,0 +1,15 @@
+class AddSourceType < ActiveRecord::Migration[5.1]
+  def change
+    create_table :source_types, :id => :bigserial do |t|
+      t.string :name
+      t.string :display_name
+      t.string :vendor
+      t.jsonb :create_schema
+      t.jsonb :update_schema
```

Comment:
I'm trying to come up with a use case for this.  Maybe a required field that is only populated on create then read-only after that?

Reply:
I wonder if we would be able to handle that with a conditional https://json-schema.org/draft-07/json-schema-validation.html#rfc.section.6.6

Reply:
We can go with just `:schema` then if one field can handle it

Reply:
I haven't tested conditionals, but I'd prefer to go with one column until we have a hard requirement for the second.